docs/admission-webhooks.md

```diff
@@ -0,0 +1,65 @@
+
+## Background
+
+OLM is capable of managing the lifecycle of [validating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#validatingadmissionwebhook) and [mutating](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) admission webhooks that are shipped alongside your operator.
+
+You may be unfamiliar with webhooks. At a high level, after a request has been authenticated and authorized, admission webhooks intercept requests against the Kubernetes API and have an opportunity to validate or update the object before it is saved in the object store. Please refer to the following table that highlights what each webhook is capable of:
+
+|                    | Validating Webhook | Mutating Webhook |
+|--------------------|--------------------|------------------|
+| Validating Objects |          x         |         x        |
+| Mutating Objects   |                    |         x        |
+
+If you are interested in learning more about admission webhooks, please review the [official kubeneretes documentation](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#what-are-they).
+
+## Defining your Webhook in the ClusterServiceVersion
+
+### Creating a Webhook
+
+It is outside the scope of this document to define the process of creating an admission webhook, however there are a few constraints that you should be aware of before developing your webhook if you plan to have it managed by OLM:
+
+#### Certificate Authority Restraints
+
+OLM is currently configured to provide each deployment with a single Certificate Authority (CA). The logic that generates and mounts the CA into the deployment was originally used by the [API Service](https://kubernetes.io/docs/tasks/access-kubernetes-api/setup-extension-api-server/) lifecycle logic. As a result:
+
+- The tls Cert file will be mounted to the deployment at `/apiserver.local.config/certificates/apiserver.crt`
+- The tls Key file will be mounted to the deployment at `/apiserver.local.config/certificates/apiserver.key`
+
+#### Webhook Rules Restrictions
+
+Additionally, in an attempt to prevent operator from configuring the cluster into an unrecoverable state, OLM does not allow the Rules defined in the WebhookDescription to:
+
+- Intercept requests that target all groups
+- Intercept requests that target the `operators.coreos.com` group
+- Intercept requests that target the `ValidatingWebhookConfigurations` or `MutatingWebhookConfigurations` resources
+
+### Defining a Webhook in a CSV
+
+OLM's ClusterServiceVersion resource was recently updated to include a [WebhookDefinition object](https://github.com/operator-framework/api/blob/master/pkg/operators/v1alpha1/clusterserviceversion_types.go#L164-L180) which can be used to define validating and mutating admission webhooks that will be shipped with the operator. For your convenience, the WebhookDefinition object can be seen below:
+
+```golang
```

Comment:
I imagine for readers a yaml example would be more beneficial than the go struct

Reply:
Addressed in the latest version of the PR.